tests/LanguageServerRename/OffsetExtractor.php

```diff
@@ -42,7 +56,7 @@ public function registerRange(string $name, string $openMarker, string $closeMar
         return $this;
     }
 
-    public function parse(string $source): array
+    public function parse(string $source): OffsetExtractor
```

Comment:
I would consider making this the static constructor `create($source)`, replacing the existing constructor. This would make the class immutable. The disadvantage is that we can't reuse the config.

Another option would be to return a new object which just has the points and ranges.

But, let's go with this unless you want to try changing it. In anycase, I think it can be refactored without changing the API.